lib/Cron/Task.php

```diff
@@ -32,12 +109,116 @@ public function __construct() {
 	 * @return void
 	 */
 	protected function run($argument) {
-		if (!\OCP\App::isEnabled('files_antivirus')) {
+		if (!\OCP\App::isEnabled('files_antivirus')
+			|| $this->appConfig->getAvScanBackground() !== 'true'
+		) {
+			return;
+		}
+
+		// locate files that are not checked yet
+		try {
+			$result = $this->getFilesForScan();
```

Comment:
It might be worthy to return the results sorted by user in order to reduce the number of times we need to switch the FS. Something to be checked later.

Reply:
Previous scan time should have precedence anyway as cron could stuck on the set of frequently changed files otherwise.